packages/context/src/resolver.ts

```diff
@@ -54,6 +54,21 @@ export function instantiateClass<T>(ctor: Constructor<T>, ctx: Context): T | Pro
 }
 
 /**
+ * Resolve the value or promise for a given injection
+ * @param ctx Context
+ * @param injection Descriptor of the injection
+ */
+function resolve<T>(ctx: Context, injection: Injection): ValueOrPromise<T> {
+  if (injection.resolve) {
+    // A custom resolve function is provided
+    return injection.resolve(ctx, injection);
+  }
+  // Default to resolve the value from the context by binding key
```

Comment:
// Resolve the value from the context via the binding key by default
// Default to resolving the value from the context via the binding key